libraries/botbuilder-core/src/telemetryLoggerMiddleware.ts

```diff
@@ -191,14 +191,14 @@ export class TelemetryLoggerMiddleware implements Middleware {
         properties[this.telemetryConstants.fromIdProperty] = activity.from ? activity.from.id : '';        
         properties[this.telemetryConstants.conversationNameProperty] = activity.conversation.name || '';        
         properties[this.telemetryConstants.localeProperty] = activity.locale || '';
-        properties[this.telemetryConstants.recipientIdProperty] = activity.recipient.id;
-        properties[this.telemetryConstants.recipientNameProperty] = activity.recipient.name;
+        properties[this.telemetryConstants.recipientIdProperty] = (activity && activity.recipient && activity.recipient.id) ? activity.recipient.id : '';
+        properties[this.telemetryConstants.recipientNameProperty] = (activity && activity.recipient && activity.recipient.name) ? activity.recipient.name : '';
```

Comment:
This will blow up on the first line if activity is null or undefined:

```ts
properties[this.telemetryConstants.fromIdProperty] = activity.from ? activity.from.id : ''; 
```

Another way of looking at it is that there isn't anything to log if there is no Activity, correct?

@garypretty can you confirm if we should add an initial `activity !== null && activity !== undefined` check to the following protected methods?

`onReceiveActivity()`
`fillReceiveEventProperties()`

Reply:
@stevengum we have a null check for activity before we call onReceiveActivity (which in turn calls fillReceiveEventProperties), so activity should never be null or undefined. https://github.com/microsoft/botbuilder-js/blob/73a63b52810071684991443d8a7ef6f5008eb51d/libraries/botbuilder-core/src/telemetryLoggerMiddleware.ts#L73-L79

Reply:
@garypretty I see your point; additionally the code here will also be in parity with C#. 

However, I think there's a flaw in both languages as all the methods in C# and TypeScript are virtual methods and public or protected.

If I override onTurn and then call the original onReceiveActivity, onReceiveActivity and fillReceiveEventProperties in the base class do not have the guarantee that context.activity is not null.

Reply:
Good catch - I can update both to check for null.